### PR TITLE
EVA-2979 Shelve duplicate ss

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

File: tasks/eva_2979/move_duplicates_to_another_collection.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+def parse_duplicate_list(duplicates_file):
+    ignored_type = ('In_original_assembly,Same_variants', 'Remapped,Same_variants,Same_variants_in_source')
+    duplicated_accessions = []
+    assembly_accession = None
+    with open(duplicates_file) as open_file:
+        for line in open_file:
+            sp_line = line.strip().split()
+            if sp_line[2] not in ignored_type:
+                duplicated_accessions.append(int(sp_line[0]))
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2979/move_duplicates_to_another_collection.py:61]: I would suggest casting as long if possible...

File: tasks/eva_2979/move_duplicates_to_another_collection.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    parser.add_argument('--profile', default='development')
+    args = parser.parse_args()
+    mongo_uri = get_mongo_uri_for_eva_profile(args.profile, args.settings_xml_file)
+    with MongoClient(mongo_uri) as mongo_handle:
+        duplicated_accessions, assembly_accession = parse_duplicate_list(args.duplicates_file)
+        if duplicated_accessions:
+            logger.info(f'Will attempt to shelf {len(duplicated_accessions)} for assembly {assembly_accession}')
+            shelve_submitted_variant_entities(mongo_handle, duplicated_accessions, assembly_accession)
+        else:
+            logger.info(f'No suplidated variants for assembly {assembly_accession}')
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2979/move_duplicates_to_another_collection.py:81]:
```suggestion
            logger.info(f'No duplicated variants for assembly {assembly_accession}')
```

File: tasks/eva_2979/move_duplicates_to_another_collection.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def parse_duplicate_list(duplicates_file):
+    ignored_type = ('In_original_assembly,Same_variants', 'Remapped,Same_variants,Same_variants_in_source')
+    duplicated_accessions = []
+    assembly_accession = None
+    with open(duplicates_file) as open_file:
+        for line in open_file:
+            sp_line = line.strip().split()
+            if sp_line[2] not in ignored_type:
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2979/move_duplicates_to_another_collection.py:60]:
```suggestion
            if sp_line[2].lower() not in ignored_type.lower():
```

File: tasks/eva_2979/move_duplicates_to_another_collection.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    batch_size = 1000
+    for batch_sve_accs in grouper(submitted_variant_accession, batch_size):
+        # remove None
+        batch_sve_accs = [acc for acc in batch_sve_accs if acc]
+        query_filter = {'seq': assembly_accession, 'accession': {'$in': batch_sve_accs}}
+        documents = [d for d in mongo_handle["eva_accession_sharded"]['dbsnpSubmittedVariantEntity'].find(query_filter)]
+        logger.info(f'Found {len(documents)} documents for {len(batch_sve_accs)} accessions')
+        duplicated_submitted_variant_ids = check_submitted_variant_have_duplicates(documents)
+        if duplicated_submitted_variant_ids:
+            mongo_handle["eva_accession_sharded"][output_collection].insert_many(duplicated_submitted_variant_ids)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2979/move_duplicates_to_another_collection.py:48]: Suggest using writeConcern majority for this and the delete below...

**Review by sundarvenkata-EBI — CHANGES_REQUESTED** [review]
Looks good... Suggest using read and write concern majority...

**Review by nitin-ebi — APPROVED** [review]

File: tasks/eva_2979/move_duplicates_to_another_collection.py (hunk 1 above)
Comment (tcezard) [reply tasks/eva_2979/move_duplicates_to_another_collection.py:68]: `int` is already a long

File: tasks/eva_2979/move_duplicates_to_another_collection.py (hunk 2 above)
Comment (tcezard) [reply tasks/eva_2979/move_duplicates_to_another_collection.py]: Done

File: tasks/eva_2979/move_duplicates_to_another_collection.py (hunk 4 above)
Comment (tcezard) [reply tasks/eva_2979/move_duplicates_to_another_collection.py]: Done

**Review by sundarvenkata-EBI — APPROVED** [review]

**Pull request merged**